### ISSUE TITLE
[MPS] tanh/sigmoid/silu backward produces wrong gradients on permuted tensors

### ISSUE THREAD

**Issue by npinto** [pytorch/pytorch#175188]
### 🐛 Describe the bug

Several activation functions produce correct forward results on non-contiguous MPS tensors but return wrong gradients during backward. I ran into this while training a model with permuted inputs on MPS.

```python
import torch

torch.manual_seed(42)
base = torch.randn(1, 7, 4, 3)
cpu_x = base.clone().permute(0, 3, 1, 2).requires_grad_(True)
mps_x = base.to("mps").clone().permute(0, 3, 1, 2).requires_grad_(True)

# forward is fine
cpu_fwd = torch.tanh(cpu_x)
mps_fwd = torch.tanh(mps_x)
torch.mps.synchronize()
print("forward diff:", (cpu_fwd.detach() - mps_fwd.detach().cpu()).abs().max().item())

# backward is wrong
cpu_fwd.sum().backward()
mps_fwd.sum().backward()
torch.mps.synchronize()
print("grad diff:", (cpu_x.grad - mps_x.grad.cpu()).abs().max().item())
```

Output:
```
forward diff: 0.0
grad diff: 1.9994
```

Same pattern with `torch.sigmoid`, `F.silu`, `F.mish` (on channels_last), `F.softplus`, `F.threshold`, and `torch.special.logit`. `F.relu` is affected only on permuted tensors, not on `moveaxis` ones.

`F.elu`, `F.gelu`, `F.leaky_relu`, `F.selu` all work fine on the same non-contiguous inputs.

I also noticed that even when the input is contiguous, these backward kernels produce wrong gradients if the incoming `grad_output` is non-contiguous (e.g. from an upstream permute).

`.contiguous()` before the op works around it.

Reproduces on 2.10.0 and nightly 2.12.0.dev20260217. CUDA is not affected.

Related: #174345, #131736

### Versions

```
Collecting environment information...
PyTorch version: 2.10.0
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 14.8.3 (arm64)
GCC version: Could not collect
Clang version: 16.0.0 (clang-1600.0.26.6)
CMake version: version 4.2.3
Libc version: N/A

Python version: 3.13.11 (main, Dec  5 2025, 16:06:33) [Clang 16.0.0 (clang-1600.0.26.6)] (64-bit runtime)
Python platform: macOS-14.8.3-arm64-arm-64bit-Mach-O
Is CUDA available: False
Is XPU available: False
Is XNNPACK available: True

CPU:
Apple M2 Max
```

Also reproduces on nightly 2.12.0.dev20260217 (same machine).

cc @kulinseth @malfet @DenisVieriu97 @jhavukainen @aditvenk
